### PR TITLE
fix: lost model data if LoadPolicy failed

### PR DESCRIPTION
**Pull request by closetool**
Signed-off-by: closetool <c299999999@qq.com>

Fix: #817

Fix: #791

### CODE REVIEW

**Comment by hsluoyz** [comment]
@elissa2333 @hiteshsethi plz review

**Review by nodece — APPROVED** [review]
LGTM

**Comment by closetool** [comment]
@hsluoyz @elissa2333 @hiteshsethi

File: enforcer.go [hunk 1]
```diff
@@ -271,20 +271,21 @@ func (e *Enforcer) ClearPolicy() {
 
 // LoadPolicy reloads the policy from file/database.
 func (e *Enforcer) LoadPolicy() error {
+	built := false
```
Comment (hsluoyz) [inline enforcer.go:274]: Change to `needToRebuild`

File: model/assertion.go [hunk 2]
```diff
@@ -96,3 +96,24 @@ func (ast *Assertion) setLogger(logger log.Logger) {
...
+	newAst := &Assertion{
+		Key:           ast.Key,
+		Value:         ast.Value,
+		PolicyMap:     policyMap,
+		Tokens:        tokens,
+		Policy:        policy,
+		priorityIndex: ast.priorityIndex,
+	}
+	return newAst
+}
```
Comment (hsluoyz) [inline model/assertion.go:119]: new line

File: model/model.go [hunk 3]
```diff
@@ -271,3 +271,16 @@ func (model Model) ToText() string {
...
+	for sec, m := range model {
+		newAstMap := make(AssertionMap)
+		for ptype, ast := range m {
+			newAstMap[ptype] = ast.copy()
+		}
+		newModel[sec] = newAstMap
+	}
+	newModel.SetLogger(model.GetLogger())
+	return newModel
+}
```
Comment (hsluoyz) [inline model/model.go:286]: new line

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Pull request merged**